src/VisualStudio/Core/Def/StackTraceExplorer/StackTraceExplorerToolWindow.cs

```diff
@@ -54,7 +56,13 @@ public StackTraceExplorerToolWindow() : base(null)
         /// </summary>
         public async Task<bool> ShouldShowOnActivatedAsync(CancellationToken cancellationToken)
         {
-            if (Root is null)
+            if (Root is null || _workspace is null)
+            {
+                return false;
+            }
+
+            var enabled = _workspace.CurrentSolution.Options.GetOption(StackTraceExplorerOptions.OpenOnFocus);
```

Comment:
Is it possible to check this earlier, before the JTF.Run dance?

Alternatively, can we unadvise the broadcast events, and re-advise, when this option is toggled? In case you can't tell yet,  putting code in WM_ACTIVATE scares me :)

Reply:
Good idea. Maybe hooking to change events would be best